modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/package.scala

```diff
@@ -7,31 +7,68 @@ import cats.Applicative
 import cats.effect._
 import cats.implicits._
 import edu.gemini.aspen.giapi.commands.Command
+import edu.gemini.aspen.giapi.status.{StatusHandler, StatusItem}
+import edu.gemini.aspen.giapi.statusservice.{StatusHandlerAggregate, StatusService}
 import edu.gemini.aspen.giapi.util.jms.status.StatusGetter
 import edu.gemini.aspen.gmp.commands.jms.client.CommandSenderClient
 import edu.gemini.giapi.client.commands.CommandResult
 import edu.gemini.jms.activemq.provider.ActiveMQJmsProvider
-import fs2.async
+import fs2.{Stream, async}
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.Duration
 
 package object client {
 
   /**
     * Allowed types according to GIAPI
     */
-  implicit val strItemGetter: ItemGetter[String]    = new ItemGetter[String] {}
-  implicit val doubleItemGetter: ItemGetter[Double] = new ItemGetter[Double] {}
-  implicit val intItemGetter: ItemGetter[Int]       = new ItemGetter[Int]    {}
-  implicit val floatItemGetter: ItemGetter[Float]   = new ItemGetter[Float]  {}
+  implicit val strItemGetter: ItemGetter[String] = new ItemGetter[String] {
+    def value(p: Any): Option[String] = p match {
+      case x: String => Some(x)
+      case _         => None
+    }
+  }
+
+  implicit val doubleItemGetter: ItemGetter[Double] = new ItemGetter[Double] {
+    def value(p: Any): Option[Double] = p match {
+      case x: Double => Some(x)
+      case _         => None
+    }
+  }
+
+  implicit val intItemGetter: ItemGetter[Int] = new ItemGetter[Int] {
+    def value(p: Any): Option[Int] = p match {
+      case x: Int => Some(x)
+      case _      => None
+    }
+  }
+
+  implicit val floatItemGetter: ItemGetter[Float] = new ItemGetter[Float] {
+    def value(p: Any): Option[Float] = p match {
+      case x: Float => Some(x)
+      case _        => None
+    }
+  }
 }
 
 package client {
 
-  import scala.concurrent.duration.Duration
-
   /**
     * Typeclass to present as evidence when calling `Giapi.get`
     */
-  sealed trait ItemGetter[A]
+  sealed trait ItemGetter[A] {
+
+    /**
+      * Attempt to convert any value to A as sent by StatusHandler
+      */
+    def value(p: Any): Option[A]
```

Comment:
You can implement this, actually.

```scala
def value(p: Any)(implicit ev: shapeless.Typeable[A]): Option[A] =
  ev.cast(p)
```

Or capture as a class constraint `abstract class ItemGetter[A: Typeable] {` which might work better.

`Typeable` instances aren't available for all types (arrays in particular) but if they are available it's safe.

Reply:
Nice, i'll try that

Reply:
Can you expand on what exactly `Typeable` does. i'm always a bit hestitant about shapeless' magic

Reply:
For your use case here it's doing more or less what you were doing. For primitive types you can look at the runtime value and figure out the type. The source is [here](https://github.com/milessabin/shapeless/blob/master/core/src/main/scala/shapeless/typeable.scala) … there's some complexity in dealing with parameterized types but we're not doing that here as far as I can tell.

Reply:
We could define our own mini-typeable but Shapeless is there and it works so I think we might as well use it.